File: Python/timus/1201segden.py
#!/usr/bin/env python3
def daynum(d,m,y):
    r=d;
    for i in range(1600,y):
        r+=365
        if i%4==0:
            if i%100!=0:
                r+=1
            elif i%400==0:
                r+=1
    for i in range(1,m):
        if i==1 or i==3 or i==5 or i==7 or i==8 or i==10 or i==12:
            r+=31
        elif i==4 or i==6 or i==9 or i==11:
            r+=30
        elif i==2:
            r+=28
            if y%4==0:
                if y%100!=0:
                    r+=1
                elif y%400==0:
                    r+=1
    return r
def mdays(m,y):
    r=0
    if m==1 or m==3 or m==5 or m==7 or m==8 or m==10 or m==12:
        r=31
    elif m==4 or m==6 or m==9 or m==11:
        r=30
    elif m==2:
        r=28
        if y%4==0:
            if y%100!=0:
                r+=1
            elif y%400==0:
                r+=1
    return r
def sstr(n):
    if n<1 or n>dm:
        return spr*2
    if n<10:
        return spr+str(n)
    return str(n)
ls=[int(it) for it in input().split(" ")]
d,m,y=ls[0],ls[1],ls[2]
s=(daynum(1,m,y)-3)%7-1
dm=mdays(m,y)
w=-((-s-dm-1)//7)
c=["mon","tue","wed","thu","fri","sat","sun"]
leng=1
cd=0
spr=" "
for i in range(7):
    c[i]+=spr
for i in range(w):
    #if 7*(i+1)-1-s>=10:
        #leng=2
    if i>0:
        leng=2
    for j in range(7):
        cd=7*i+j-s
        if cd==d:
            c[j]+=spr*(leng-1)+"["+sstr(cd)+"]"
        else:
            c[j]+=spr*leng+sstr(cd)+spr
for i in range(7):
    if c[i][-1]==spr:
        c[i]=c[i][:-1]
for strg in c:
    print(strg)#+"e"

File: C/timus/1201segden.c
#include <stdio.h>
int daynum(int d,int m,int y) {
    long r=d;
    int i;
    for (i=1600;i<y;i++) {
        r+=365;
        if ((!(i%4))&&(i%100))
            r+=1;
        if (!(i%400))
            r+=1;
    }
    for (i=1;i<m;i++) {
        if ((i==1)||(i==3)||(i==5)||(i==7)||(i==8)||(i==10)||(i==12))
            r+=31;
        if ((i==4)||(i==6)||(i==9)||(i==11))
            r+=30;
        if (i==2) {
            r+=28;
            if ((!(y%4))&&(y%100))
                r+=1;
            if (!(y%400))
                r+=1;
        }
    }
    return r;
}
int mdays(int m,int y) {
    int r;
    if ((m==1)||(m==3)||(m==5)||(m==7)||(m==8)||(m==10)||(m==12))
        r=31;
    if ((m==4)||(m==6)||(m==9)||(m==11))
        r=30;
    if (m==2) {
        r=28;
        if ((!(y%4))&&(y%100))
            r+=1;
        if (!(y%400))
            r+=1;
    }
    return r;
}
int bdiv(int a,int b) {
    int r=a/b;
    if (a%b)
        r++;
    return r;
}
int main() {
    int d,m,y,i,j,t;
    scanf("%d%d%d",&d,&m,&y);
    int s=(daynum(1,m,y)+4)%7-1,dm=mdays(m,y),w=bdiv(s+dm+1,7),len=5*w+2,cd;
    //printf("%d %d %d %d\n",s,dm,w,len);
    char c[7][50]={"mon","tue","wed","thu","fri","sat","sun"},buf[10],sep=' ';
    for (i=0;i<7;i++) {
        for (j=3;j<len;j++)
            c[i][j]=sep;
        c[i][len]='\0';
        c[i][len+1]='\0';
        //printf("%s\n",&c[i]);
    }
    for (i=0;i<w;i++) {
        for (j=0;j<7;j++) {
            cd=7*i+j-s;
            if ((cd>0)&&(cd<=dm)) {
                sprintf(&c[j][5*i+5],"%2d",cd);
                if (i!=w-1)
                    c[j][5*i+7]=sep;
                //printf("%2d\n",cd);
                //sprintf(buf,"%2d",cd);
                if (cd==d) {
                    c[j][5*i+4]='[';
                    c[j][5*i+7]=']';
                }
            }
        }
    }
    for (i=0;i<7;i++)
        printf("%s\n",&c[i]);
    return 0;
}